src/mbgl/gl/context.cpp

```diff
@@ -248,10 +248,16 @@ UniqueTexture Context::createTexture() {
 }
 
 bool Context::supportsVertexArrays() const {
-    return vertexArray &&
-           vertexArray->genVertexArrays &&
-           vertexArray->bindVertexArray &&
-           vertexArray->deleteVertexArrays;
+    // Blacklist Adreno 3xx as it crashes on glBuffer(Sub)Data
+    const std::string renderer = reinterpret_cast<const char*>(glGetString(GL_RENDERER));
+    if (renderer.find("Adreno (TM) 3") != std::string::npos) {
```

Comment:
Can you make part of this check static? Its called on every every call to `Context::createVertexArray()`